packages/tasit-action/src/NFT.test.js

```diff
@@ -35,4 +90,69 @@ describe("TasitAction.NFT", () => {
     expect(name).to.exist;
     expect(name).to.equal("Full NFT");
   });
+
+  describe("ERC721 functions", () => {
+    const tokenId = 1;
+
+    beforeEach("should mint one token to owner", async () => {
+      const balanceBefore = await fullNFT.balanceOf(owner.address);
+      expect(toBN(balanceBefore)).to.be.bignumber.equal(toBN(0));
+
+      txSubscription = fullNFT.mint(owner.address, tokenId);
+      await txSubscription.waitForNonceToUpdate();
```

Comment:
I've attempted to not test so much event here since we have done that on Contract.test.js but I'll follow your suggestion.

Reply:
The reason I bring it up is because I wasn't sure if the nonce having updated would necessarily mean that the transaction was confirmed and therefore that the state had changed, so I thought this could lead to intermittent test failures

Reply:
Rethink about that:
`waitForNonceToUpdate` <=> `waitForTxConfirmation` <=> `waitForNextBlock`
So, after this process wallet nonce is synced with expected nonce for next transaction.

> In practical terms, the nonce is an up-to-date count of the number of confirmed (i.e., on-chain) transactions that have originated from an account. To find out what the nonce is, you can interrogate the blockchain, for example via the web3 interface. Open a JavaScript console in a browser with MetaMask running, or use the truffle console command to access the JavaScript web3 library, then type:

Refs: https://github.com/ethereumbook/ethereumbook/blob/04f66ae45cd9405cce04a088556144be11979699/06transactions.asciidoc#keeping-track-of-nonces

Reply:
Ah ok that does help to clear it up. I guess conceivably there could be a block that didn't include the tx, though. But the first two being the same makes sense.

Reply:
Resolving this